### DISCUSSION TITLE
Patchcore Tiling is False in Config.Yaml

### DISCUSSION THREAD

**Discussion by nicozhou7** [open-edge-platform/anomalib#735] [Q&A]
Hi, I saw the patchcore config.yaml is having tiling=false. If this is false, does this means it will not split the image to patch and directly input whole image for feature extraction? Does this means it is different from paper implementation?

**To Reproduce**

1. Go to anomalib/patchcore/config.yaml
2. See the tiling is set to false

**Hardware and Software Configuration**

- OS: [Ubuntu, OD]
- NVIDIA Driver Version [470.57.02]
- CUDA Version [e.g. 11.4]
- CUDNN Version [e.g. v11.4.120]
- OpenVINO Version [Optional e.g. v2021.4.2]

**Comment by djdameln** [comment]
Hi, thanks for your question. The tiling section in the `config.yaml` has nothing to do with the patching of the input image in the Patchcore implementation.

In Anomalib, we provide an extra, optional tiling mechanism which consists of splitting the input image into smaller tiles before feeding the images into the model. Instead of resizing the full image to the model's input size, we first subdivide the image into a grid of (e.g. 4x4) tiles, after which we resize the individual tiles to the input size of the model. This allows us to preserve more of the original image resolution, which may help in the detection of small anomalies. After the model makes its predictions on the individual tiles, the tiled predictions are stitched back together so that the output has the shape of the original image.

Please note that this tiling is done as a pre- and post-processing step and is independent of the workings of any of the models in Anomalib. So when tiling is disabled, the PatchCore model should behave exactly as the paper implementation.

**Reply by nicozhou7** [reply]
may I know where does this splitting the input image to patch size 4x4? Cause I want to understand it further and how can I change the patchsize? Cause the original implementation has this

**Reply by nicozhou7** [reply]
Thanks for the answer, but may I know where does the patching in patchcore implemented? Cause the original code has the patchsize parameters and I want to try to change it.

**Reply by nicozhou7** [reply]
` 
 


      if self.tiler:
            input_tensor = self.tiler.tile(input_tensor)


        with torch.no_grad():
            features = self.feature_extractor(input_tensor)

        features = {layer: self.feature_pooler(feature) for layer, feature in features.items()}
        embedding = self.generate_embedding(features)

        if self.tiler:
            embedding = self.tiler.untile(embedding)

        batch_size, _, width, height = embedding.shape
        embedding = self.reshape_embedding(embedding)

`

**Reply by nicozhou7** [reply]
so what is the self.tiler.tile() do? it split tensor for input to patchcore model isnt it?
